android/app/src/androidTest/java/PrnmTest.java

```diff
@@ -0,0 +1,303 @@
+// Copyright (c) 2021 Chair of Applied Cryptography, Technische Universität
+// Darmstadt, Germany. All rights reserved. This file is part of
+// perun-eth-mobile. Use of this source code is governed by the Apache 2.0
+// license that can be found in the LICENSE file.
+
+package network.perun.app;
+
+import prnm.*;
+
+import java.math.*;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicReference;
+
+import android.util.Log;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.jupiter.api.Assertions;
+import androidx.test.runner.AndroidJUnit4;
+import androidx.test.core.app.ApplicationProvider;
+import static com.google.common.truth.Truth.assertThat;
+import static org.awaitility.Awaitility.*;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+
+@RunWith(AndroidJUnit4.class)
+public class PrnmTest implements prnm.NewChannelCallback, prnm.ProposalHandler, prnm.UpdateHandler, prnm.ConcludedEventHandler {
+
+    /**
+     * This class contains four entry points for the test-runner; two per peer and test case.
+     *
+     * runFirst* sets up a channel, sends TXs and persists it.
+     * runSecond* asserts that the channel can be restored from persistence.
+     */
+
+    Client client;
+    Context ctx = Prnm.contextWithTimeout(120);
+    String prefix;
+    Setup setup;
+
+    AtomicInteger receivedTx = new AtomicInteger(0);
+    AtomicBoolean watcherStopped = new AtomicBoolean(false);
+    // handleConcluded returned
+    AtomicBoolean concluded = new AtomicBoolean(false);
+    AtomicReference<PaymentChannel> ch = new AtomicReference<PaymentChannel>(null);
+
+    @Test
+    /**
+     * Alice entry point for the first test.
+     */
```

Comment:
I would either move the `@Test` tag after the Javadoc comment or replace it with a simple one-line // comment since it is not really a Javadoc comment but rather a short remark for the method.  (The same for the other entry points).

Reply:
Ok, i was not aware that there is a difference in `/**` and `//` comments in java.